ledger/lruaccts.go

```diff
@@ -117,5 +152,8 @@ func (m *lruAccounts) prune(newSize int) (removed int) {
 		m.accountsList.remove(back)
 		removed++
 	}
+
+	// clear the notFound list
+	m.notFound = make(map[basics.Address]struct{}, len(m.notFound))
```

Comment:
Supposedly, deleting in a loop over all keys is recognized by the compiler and optimized to be cheaper than reallocating. @cce posted something about it. https://go-review.googlesource.com/c/go/+/110055